platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationFilesPoller.java

```diff
@@ -199,4 +130,82 @@ private void processExistingConfigurationFiles() {
     Stream<Path> getExistingFiles() throws IOException {
         return Files.walk(configurationDirectoryPath);
     }
+
+    private class ListenerImpl implements FileAlterationListener {
+        @Override
+        public void onStart(FileAlterationObserver fileAlterationObserver) {
+        }
+
+        @Override
+        public void onDirectoryCreate(File file) {
+        }
+
+        @Override
+        public void onDirectoryChange(File file) {
+        }
+
+        @Override
+        public void onDirectoryDelete(File file) {
+        }
+
+        @Override
+        public void onFileCreate(File file) {
+            try {
+                String filename = file.getName();
+                LOGGER.debug("Watcher has been notified. Handling event for [{}].", filename);
+
+                if (!filename.endsWith(fileExtension)) {
```

Comment:
❓ Couldn't we use a `IOFileFilter` when we create the `FileAlterationObserver` instead of doing this every time this method gets called?